### ISSUE TITLE
Future that wakes itself monopolizes thread-local executor.

### ISSUE THREAD

**Issue by Dirbaio** [stjepang/smol#171]
Test case:

```rust
use std::pin::Pin;
use std::task::{Context, Poll, Waker};
use futures::prelude::*;

struct Greedy;
impl Future for Greedy {
    type Output = ();

    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<()> { 
        cx.waker().wake_by_ref();
        Poll::Pending
    }
}

async fn tick() {
    loop {
        println!("Tick!");
        smol::Timer::after(std::time::Duration::from_secs(1)).await;
    }
}

fn main() {
    smol::run(async {
        println!("Hello!");
        
        smol::Task::local(tick()).detach();
        smol::Task::local(Greedy).detach();

        // Sleep forever.
        let () = future::pending().await;
    })
}
```

This prints `Tick!` only once, then the `tick` task never gets to run again. If you remove the `Greedy` task, `Tick!` is printed every second as expected.

It seems to me the `tick` task should get to run, since `Greedy` is constantly yielding (returning `Poll::Pending`).

smol v0.1.14

**Comment by stjepang** [comment]
Thank you for the report! This should now be fixed in v0.1.15 - can you verify?

**Comment by Dirbaio** [comment]
I confirm the fix works, both for the sample test above and the original program I found it in.

Thanks a lot for the fast fix!
